pybamm/expression_tree/broadcasts.py

```diff
@@ -45,6 +45,27 @@ def _sympy_operator(self, child):
         """Override :meth:`pybamm.UnaryOperator._sympy_operator`"""
         return child
 
+    def diff(self, variable):
+        """
+        Override :meth:`pybamm.SpatialOperator.diff()` to reinstate behaviour of
+        :meth:`pybamm.Symbol.diff()`.
+        """
```

Comment:
probably cleaner to just get rid of the NotImplementedError for spatial operators